### ISSUE TITLE
Using the "Robot" class to clamp the Mouse can freeze the entire OS

### ISSUE THREAD

**Issue by steffen-wilke** [gurkenlabs/litiengine#263]
Recently, I've had a weird bug where my system would sometimes freeze entirely when a LITIengine game was executed. This occurred entirely random and there was no reliable way to reproduce it. I've done some research during the LDJAM and found out that using the **`Robot`** class can under certain circumstances cause this behavior. The engine uses this class to clamp the `Mouse` to the game window to prevent it from leaving the application.
After disabling the clamping via `Input.mouse().setGrabMouse(false)`, the problem did no longer occur... I've observed this problem irregularly for a few weeks to months now... at first, I thought it might somehow be related to my new machine. The engine didn't change anything related to how it uses the `Robot` class recently, something external must have changed.

**Some ideas about the issue:**
* Maybe it's related to the Mouse refresh rate
* Maybe the OS changed its handling for some input-related stuff 
* Maybe the JVM has a problem  -> re-installing didn't help
* Similar issue: https://bugs.openjdk.java.net/browse/JDK-8196030?attachmentOrder=desc

We need to do some research to solve this.

**Specs:**
* **CPU:** Core i7-8700K - 3.70 GHz
* **OS:** Windows 10 Pro - 10.0.17763 Build 17763
* **Mouse:** Logitech G903 - 1000hz 1600DPI
* **JVM:** Java SE 8u201

**Comment by TheRamenChef** [comment]
Why does the engine clamp the mouse, anyway? It's a 2D engine.

**Comment by steffen-wilke** [comment]
I don't think it's a matter of 2D vs 3D. If you have a game where you control the player or something else  with the mouse (e.g. MOBA or Strategy Game controls) not clamping the mouse can often-times result in unwanted behavior, especially in full-screen mode. I agree though, that maybe this shouldn't be the default.

**Comment by TheRamenChef** [comment]
I would think that mouse controls would be inhibited by clamping. Isn't the standard way of scrolling around a large map to move the mouse near the edge of the screen?

What unwanted behavior would result?

**Comment by steffen-wilke** [comment]
Okay, I'm not 100% sure if "clamping" is the right term I used... What I mean is: To ensure that the mouse cannot leave the game window. Maybe "grabbing" or "locking" is more fitting here. 

> I would think that mouse controls would be inhibited by clamping. Isn't the standard way of scrolling around a large map to move the mouse near the edge of the screen?

In fact, we've implemented this for one of our games an the behavior was only possible due to locking the mouse to the window. Otherwise, the mouse might just go off the bounds of the window (e.g. second screen) and the player would drop back to the desktop upon clicking.
